invenio_github/views/github.py

```diff
@@ -24,225 +24,221 @@
-                    repo.latest_release())
+    return render_template(current_app.config["GITHUB_TEMPLATE_INDEX"], **ctx)
 
-        last_sync = humanize.naturaltime(
-            (utcnow() - parse_timestamp(extra_data['last_sync'])))
 
-        ctx.update({
-            'connected': True,
-            'repos': sorted(repos.items(), key=lambda x: x[1]['full_name']),
-            'last_sync': last_sync,
-        })
+@blueprint.route("/repository/<path:repo_name>")
+@request_session_token()
+def get_repository(repo_name):
+    """Displays one repository.
 
-    return render_template(current_app.config['GITHUB_TEMPLATE_INDEX'], **ctx)
-
-
-@blueprint.route('/repository/<path:name>')
-@login_required
-@register_breadcrumb(blueprint, 'breadcrumbs.settings.github.repo',
-                     _('Repository'))
-def repository(name):
-    """Display selected repository."""
+    Retrieves and builds context to display all repository releases, if any.
+    """
     user_id = current_user.id
     github = GitHubAPI(user_id=user_id)
-    token = github.session_token
-
-    if token:
-        repos = github.account.extra_data.get('repos', [])
-        repo = next((repo for repo_id, repo in repos.items()
-                     if repo.get('full_name') == name), {})
-        if not repo:
-            abort(403)
-
-        try:
-            # NOTE: Here we do not check for repository ownership, since it
-            # might have changed even though the user might have made releases
-            # in the past.
-            repo_instance = Repository.get(user_id=user_id,
-                                           github_id=repo['id'],
-                                           check_owner=False)
-        except RepositoryAccessError:
-            abort(403)
-        except NoResultFound:
-            repo_instance = Repository(name=repo['full_name'],
-                                       github_id=repo['id'])
-
-        releases = [
-            current_github.release_api_class(r) for r in (
-                repo_instance.releases.order_by(db.desc(Release.created)).all()
-                if repo_instance.id else []
-            )
-        ]
+
+    try:
+        # NOTE: Here we do not check for repository ownership, since it
+        # might have changed even though the user might have made releases
+        # in the past.
+        repo = github.get_repository(repo_name)
+        releases = github.get_repository_releases(repo=repo)
         return render_template(
-            current_app.config['GITHUB_TEMPLATE_VIEW'],
-            repo=repo_instance,
-            repo_info=repo,
+            current_app.config["GITHUB_TEMPLATE_VIEW"],
+            repo=repo,
             releases=releases,
             serializer=current_github.record_serializer,
         )
+    except RepositoryAccessError as e:
+        abort(403)
+    except NoResultFound as e:
+        abort(404)
 
-    abort(403)
 
+###
+# TODO to be moved to its own folder (e.g. separate ui / api)
```

Comment:
Do we want to separate `ui` and `api` endpoints? We do this separation in some modules, it's easier to navigate the code.